CORDA-3841 Check `isAnyCheckpointPersisted` in `startFlowInternal`

lankydan opened the pull request:
Only hit the database if `StateMachineState.isAnyCheckpointPersisted`
returns true. Otherwise, there will be no checkpoint to retrieve from the
database anyway. This can prevent errors due to a transient loss of
connection to the database.

lankydan commented:
Run tests: slow

lankydan commented:
Run tests: slow

williamvigorr3 commented on node/src/main/kotlin/net/corda/node/services/persistence/DBCheckpointStorage.kt:
```diff
@@ -66,6 +66,7 @@ class DBCheckpointStorage : CheckpointStorage {
         return session.createQuery(delete).executeUpdate() > 0
     }
 
+    @Throws(SQLException::class)
```
Is this needed by the byteman tests somehow?

lankydan replied:
Yes.

lankydan committed:
CORDA-3841 Check `isAnyCheckpointPersisted` in `startFlowInternal`

Only hit the database if `StateMachineState.isAnyCheckpointPersisted`
returns true. Otherwise there will be no checkpoint to retrieve from the
database anyway. This can prevent errors due to a transient loss of
connection to the database.
